docs/source/learn/trace-vs-execution-time.md

```diff
@@ -0,0 +1,105 @@
+# Tracing Time vs. Execution Time in PyTorch/XLA
+
+When working with PyTorch/XLA, it's essential to understand that operations on
+XLA tensors are not typically executed immediately in the way they are with
+standard PyTorch tensors on CPU or CUDA devices (which operate in "eager mode").
+PyTorch/XLA employs a "lazy execution" model. This means that when you write
+PyTorch code using XLA tensors, you are primarily defining or tracing a
+computation graph. The compilation of the currently traced graph and it's
+subsequent execution on the device is deferred until a specific trigger point.
+
+This leads to two distinct types of "time" to consider:
+
+1. Host-Side Time: The period during which your CPU (host) prepares the
+computation. This includes:
+
+   - Tracing Time: The period during which PyTorch/XLA records your operations
+   and builds the computation graph.
+
+   - Compilation Time: The time the host-side XLA compiler takes to transform
+   the traced graph into optimized device code. This is most significant on the
+   first execution of a new graph or if the graph changes.
+
+2. Device Time: The period during which the XLA device (e.g., TPU) actually
+performs the computations. This is primarily:
+
+   - Execution Time: The time the XLA device spends running the compiled code.
```

Comment:
I don't understand why it's worth calling out "time spent running the compiled code" as a sub-part of "[time spent] perform[ing] the computation"

Reply:
merged the two into one. PTAL